firefox-ios/Client/Coordinators/TabTray/TabTrayCoordinator.swift

```diff
@@ -3,27 +3,32 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
 import Common
+import Storage
 
 protocol TabTrayCoordinatorDelegate: AnyObject {
     func didDismissTabTray(from coordinator: TabTrayCoordinator)
 }
 
 protocol TabTrayNavigationHandler: AnyObject {
     func start(panelType: TabTrayPanelType, navigationController: UINavigationController)
+    func shareTab(url: URL, sourceView: UIView)
 }
 
 class TabTrayCoordinator: BaseCoordinator,
                           ParentCoordinatorDelegate,
                           TabTrayViewControllerDelegate,
                           TabTrayNavigationHandler {
     private var tabTrayViewController: TabTrayViewController!
-    private var profile: Profile
+    private let profile: Profile
+    private let tabManager: TabManager
     weak var parentCoordinator: TabTrayCoordinatorDelegate?
 
     init(router: Router,
          tabTraySection: TabTrayPanelType,
-         profile: Profile) {
+         profile: Profile,
+         windowManager: WindowManager = AppContainer.shared.resolve()) {
         self.profile = profile
+        self.tabManager = windowManager.tabManager(for: windowManager.activeWindow)
```

Comment:
Ideally I think we'd want to inject the `tabManager` or the window UUID, rather than use `activeWindow`. The current UUID is a property on either the SceneCoordinator or BrowserCoordinator (IIRC) so it should be available at this point (or can be injected). If not, can you please add a `//TODO` here for me? I can take a closer look at `TabTrayCoordinator` next for multi-window.

Reply:
@OrlaM I added a task to revisit this so you can disregard previous comment if you prefer; I will need to take a closer look at `TabTrayCoordinator` anyway for multi-window